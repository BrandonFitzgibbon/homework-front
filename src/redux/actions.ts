import { IEntry } from '../types/interfaces'

export enum Type {
    POST_ALL_ENTRY = "POST_ALL_ENTRY",
    POST_CURRENT_ENTRY = "POST_CURRENT_ENTRY",
    SET_ALL_ENTRIES = "SET_ALL_ENTRIES",
    SET_CURRENT_ENTRIES = "SET_CURRENT_ENTRIES",
    CHANGE_NAME = "CHANGE_NAME",
    CHANGE_TARGET = "CHANGE_TARGET"
}

export function postAllEntry(entry: IEntry) {
    return { type: Type.POST_ALL_ENTRY, entries: [entry] }
}

export function postCurrentEntry(entry: IEntry) {
    return { type: Type.POST_CURRENT_ENTRY, entries: [entry] }
}

export function setAllEntries(entries: IEntry[]) {
    return { type: Type.SET_ALL_ENTRIES, entries}
}

export function setCurrentEntries(entries: IEntry[]) {
    return { type: Type.SET_CURRENT_ENTRIES, entries}
}

export function changeName(name: string) {
    return { type: Type.CHANGE_NAME, name }
}

export function changeTarget(target_id: string) {
    return { type: Type.CHANGE_TARGET, target_id}
}