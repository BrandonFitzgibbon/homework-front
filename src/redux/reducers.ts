import { combineReducers } from 'redux'
import { IEntry } from '../types/interfaces'
import { Type } from './actions'

function entries(state: IEntry[] = [], action: { type: Type, entry: IEntry }) : IEntry[] {
    switch(action.type) {
        case Type.POST_ENTRY:
            return [
                ...state,
                action.entry
            ]
        default:
            return state
    }
}

const app = combineReducers({
    entries: entries
})

export default app

