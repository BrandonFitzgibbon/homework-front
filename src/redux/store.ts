import { createStore } from 'redux'
import app from './reducers'

export const store = createStore(app, (window as any).__REDUX_DEVTOOLS_EXTENSION__ && (window as any).__REDUX_DEVTOOLS_EXTENSION__())