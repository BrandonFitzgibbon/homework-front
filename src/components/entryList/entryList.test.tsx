import React from 'react'
import EntryList from './index'
import renderer, { act } from 'react-test-renderer'
import { store } from '../../redux/store'
import { Provider } from 'react-redux'
import { postEntry } from '../../redux/actions'

test('list adds Entry when state changes', () => {
    const list = renderer.create(
        <Provider store={store}>
            <EntryList />
        </Provider>
    )
    let tree = list.toJSON()
    expect(tree).toMatchSnapshot()

    //add item to list
    act(() => {
        store.dispatch(postEntry({
            content: "hello, world"
        }))
    })

    tree = list.toJSON()
    expect(tree).toMatchSnapshot()

    //add another item to the list
    act(() => {
        store.dispatch(postEntry({
            content: "yolo"
        }))
    })

    tree = list.toJSON()
    expect(tree).toMatchSnapshot()
}) 