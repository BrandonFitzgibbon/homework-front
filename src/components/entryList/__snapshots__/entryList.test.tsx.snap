// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list adds Entry when state changes 1`] = `<ul />`;

exports[`list adds Entry when state changes 2`] = `
<ul>
  <div>
    hello, world
  </div>
</ul>
`;

exports[`list adds Entry when state changes 3`] = `
<ul>
  <div>
    hello, world
  </div>
  <div>
    yolo
  </div>
</ul>
`;
